elastic-buffer-sim/src/Bittide/ClockControl/Strategies/Callisto.hs

```diff
@@ -43,6 +45,35 @@ wrappingCounter upper = counter
   go 0 = upper
   go n = pred n
 
+-- | A version of 'sum' that is guaranteed not to overflow.
+safeSum ::
+  ( KnownNat n
+  , KnownNat m
+  , 1 <= n
+  ) =>
+  Vec n (Unsigned m) ->
+  Unsigned (m + n - 1)
+safeSum = sum . map extend
```

Comment:
I think we'll have to make make a `bittide-instance` for this component to make sure we meet timing with larger structures like these.

Reply:
Agreed. Same for `callisto`.

Reply:
I'm going to leave this to Hidde/you in a different PR :-)